# schneebergerlab/scatacseqrecomb: Output

## Introduction

This document describes the output produced by the pipeline.

The directories listed below will be created in the results directory after the pipeline has finished. All paths are relative to the top-level results directory.

<!-- TODO nf-core: Write this documentation describing your workflow's output -->

## Pipeline overview

The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes data using the following steps:

- [Cell Ranger ATAC](#cellranger) - Read alignment ([`Cell Ranger ATAC`](https://support.10xgenomics.com/single-cell-atac/software/pipelines/latest/what-is-cell-ranger-atac))
- [cellsnp-lite](#cellsnp) - Reference and alternate allele counts at specified heterozygous sites ([`cellsnp-lite`](https://cellsnp-lite.readthedocs.io/en/latest/index.html))
- [MultiQC](#multiqc) - Aggregate report describing results and QC from the whole pipeline
- [Pipeline information](#pipeline-information) - Report metrics generated during the workflow execution

### Cell Ranger ATAC

<details markdown="1">
<summary>Output files</summary>

- `cellranger/`
  - `*genome specified in reference config*`: Cell Ranger ATAC index files of the reference genome, that can be used for subsequent runs.
  - `*sample_names*`: Directory containing output files for each sample ID specified in the sample sheet.

</details>

[Cell Ranger ATAC](https://support.10xgenomics.com/single-cell-atac/software/pipelines/latest/what-is-cell-ranger-atac) align single-cell ATAC data to a reference genome and perform various analysis on it. See the [Cell Ranger ATAC](https://support.10xgenomics.com/single-cell-atac/software/pipelines/latest/what-is-cell-ranger-atac) documentation for further reading.

### cellsnp-lite

<details markdown="1">
<summary>Output files</summary>

- `cellsnp/`
  - `*sample_names*`: Directory containing output files for each sample ID specified in the sample sheet.

</details>

[`cellsnp-lite`](https://cellsnp-lite.readthedocs.io/en/latest/index.html) takes an alignment file and a VCF file containing heterozygous SNPs and counts the number of reads supporting the reference and/or the alternate allele for each site and each barcode. align single-cell ATAC data to a reference genome and perform various analysis on it. See the [cellsnp-lite](https://cellsnp-lite.readthedocs.io/en/latest/) documentation for further reading.

### Pipeline information

<details markdown="1">
<summary>Output files</summary>

- `pipeline_info/`
  - Reports generated by Nextflow: `execution_report.html`, `execution_timeline.html`, `execution_trace.txt` and `pipeline_dag.dot`/`pipeline_dag.svg`.
  - Reports generated by the pipeline: `pipeline_report.html`, `pipeline_report.txt` and `software_versions.yml`. The `pipeline_report*` files will only be present if the `--email` / `--email_on_fail` parameter's are used when running the pipeline.
  - Reformatted samplesheet files used as input to the pipeline: `samplesheet.valid.csv`.

</details>

[Nextflow](https://www.nextflow.io/docs/latest/tracing.html) provides excellent functionality for generating various reports relevant to the running and execution of the pipeline. This will allow you to troubleshoot errors with the running of the pipeline, and also provide you with other information such as launch commands, run times and resource usage.
